### PR TITLE
`PopoverPrimitive` - Allow dynamic toggle element

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

If merged, this PR will fix a bug where the `Hds::PopoverPrimitive` would stop working if its toggle element was dynamically swapped, ensuring the popover remains correctly associated and positioned.

### :hammer_and_wrench: Detailed description

The previous implementation only ran the logic to link the toggle and popover (`popovertarget`, `aria-controls`, and positioning) once when the component was first rendered. This caused a bug when a consumer dynamically changed the toggle element, as the new element was never linked to the popover.

**Example of Broken Behavior**

In the code below, clicking the "Original" button worked, but after a state change, the "New" button would not trigger the popover.

```hbs
<Hds::PopoverPrimitive @enableClickEvents={{true}} as |PP|>
  {{#if this.isSwapped}}
    <button {{PP.setupPrimitiveToggle}}>New Toggle</button>
  {{else}}
    <button {{PP.setupPrimitiveToggle}}>Original Toggle</button>
  {{/if}}
  <div {{PP.setupPrimitivePopover}}>Content</div>
</Hds::PopoverPrimitive>
```

### :link: External links

Jira ticket: [HDS-4701](https://hashicorp.atlassian.net/browse/HDS-4701)

***

### :eyes: Component checklist

- [ ] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

<details>

<summary>:clipboard: PCI review checklist</summary>

- [ ] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.
- [ ] If applicable, I've worked with GRC to document the impact of any changes to security controls.
  Examples of changes to controls include access controls, encryption, logging, etc.
- [ ] If applicable, I've worked with GRC to ensure compliance due to a significant change to the in-scope PCI environment.
  Examples include changes to operating systems, ports, protocols, services, cryptography-related components, PII processing code, etc.

</details>

### CODE REVIEW

File: packages/components/src/components/hds/popover-primitive/index.ts [hunk 1]
```diff
@@ -106,13 +107,23 @@ export default class HdsPopoverPrimitive extends Component<HdsPopoverPrimitiveSi
...
       assert(
         `The toggle element of "Hds::PopoverPrimitive" must be a <button>; element received: <${element.tagName.toLowerCase()}>`,
         element instanceof HTMLButtonElement
       );
+
+      this._linkToggleAndPopover();
+
+      // Return a teardown function to clean up the modifier's side effects.
+      // This is a safeguard against bugs where this element might be
+      // cached and re-parented in the DOM, rather than being fully destroyed.
```
Comment (didoo) [inline packages/components/src/components/hds/popover-primitive/index.ts:122]: [praise] 👏 👏 👏 👏 👏 👏

File: showcase/tests/integration/components/hds/popover-primitive/index-test.js [hunk 2]
```diff
@@ -103,6 +103,72 @@ module(
       // should go back to hidden
       assert.dom('#test-popover-primitive-content').isNotVisible();
     });
+    test('it should continue to work when the toggle element is dynamically swapped', async function (assert) {
```
Comment (didoo) [inline showcase/tests/integration/components/hds/popover-primitive/index-test.js:106]: [praise] 🧑‍🍳👌

File: .changeset/brave-owls-jump.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+<!-- START components/popover-primitive -->
+`PopoverPrimitive` - Added support for dynamic toggle elements.
```
Comment (didoo) [inline .changeset/brave-owls-jump.md]: Maybe something like this?
```suggestion
`PopoverPrimitive` - Added support for dynamic swap/injection of the toggle element.
```

File: packages/components/src/components/hds/popover-primitive/index.ts [hunk 4]
```diff
@@ -167,27 +163,58 @@ export default class HdsPopoverPrimitive extends Component<HdsPopoverPrimitiveSi
...
+    }
+  );
+
+  // Apply the `hds-anchored-position` modifier to the "popover" element
+  // (notice: this function runs the first time when the element the modifier was applied to is inserted into the DOM, and it autotracks while running.
+  // Any tracked values that it accesses will be tracked, including the arguments it receives, and if any of them changes, the function will run again)
+  // This modifiers uses the Floating UI library to provide:
+  // - positioning of the "popover" in relation to the "toggle"
+  // - collision detection (optional)
+  private _updatePositioning(): void {
```
Comment (didoo) [inline packages/components/src/components/hds/popover-primitive/index.ts]: [suggestion] `_updatePositioning` reminds me of the action that the Floating UI library is performing, while here we're actually applying the modifier to the popover element
```suggestion
  private _applyAnchoredPositionModifier(): void {
```

**Review by didoo — APPROVED** [review]
Great PR! 🙌

I have left only a few suggestions, but nothing is a blocker.

**Comment by alex-ju** [comment]
I followed the logic and I can see why this implementation is flagged as minor, which is great. Should we plan to test it with a release candidate though?

File: .changeset/brave-owls-jump.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+<!-- START components/popover-primitive -->
```
Comment (dchyun) [inline .changeset/brave-owls-jump.md]:
```suggestion
<!-- START utilities/popover-primitive -->
```
[Nit] Use correct docs path

**Review by shleewhite — COMMENTED** [review]
Could you add an example of the toggle switching to the showcase?

**Comment by didoo** [comment]
> Could you add an example of the toggle switching to the showcase?

[premise: no strong opinions] I thought the same at the beginning, but then I thought that maybe in this case it's not necessary, there is nothing UI related that a visual regression tests or a human test could catch, that the integration tests does not already catch

**Comment by zamoore** [comment]
> I followed the logic and I can see why this implementation is flagged as minor, which is great. Should we plan to test it with a release candidate though?

Independently of the next release? We can, but I don't think it's necessary.

**Comment by zamoore** [comment]
> > Could you add an example of the toggle switching to the showcase?
> 
> [premise: no strong opinions] I thought the same at the beginning, but then I thought that maybe in this case it's not necessary, there is nothing UI related that a visual regression tests or a human test could catch, that the integration tests does not already catch

@shleewhite @didoo 

This was my thinking as well. I can add the examples in the showcase if anyone has a strong opinion, but this more covers an edge case than provides additional intentional behavior.

**Comment by shleewhite** [comment]
@zamoore @didoo ehh im convinced- we can leave it out for now.

**Review by didoo — APPROVED** [review]
